### PR TITLE
Feature/linux dmg support

### PR DESCRIPTION
**Pull request by justinmchase**
This PR adds the ability to generate macos .dmg files when running on linux. Combined with the merged branch `justin/win32-support` this allows linux to build both windows and mac installers, which allows us to generate said installers from a single build on travis-ci.

This PR passes `npm test`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,11 +39,21 @@ part of package.json
...
-If you're on OS X/Linux and want to build for Windows, you need Wine installed. Wine is required in order to set the correct icon for the exe.
+If you're on OS X/Linux and want to build for Windows, you need Wine installed. Wine is required in order to set the correct icon for the exe. 
 
+You will also need the nullsoft scriptable install system for all platforms.
+
+On osx via brew
 ```
 $ brew install wine makensis
 ```
+On linux via apt-get
```
Comment (pyro2927) [inline README.md]: Following the formatting above, osx and linux should probably be capitalized: OS X & Linux.

Comment (justinmchase) [inline README.md]: Ok. I'll fix it in the other PR. This was branched off of justin/win-32 so it has some duplicate commits. When that other one gets merged in then they should drop off this PR.

Comment (stefanjudis) [inline README.md]: Just merged the other one. :bowtie:

File: lib/macos.js [hunk 2]
```diff
@@ -24,49 +117,18 @@ var MacosBuilder = {
...
-    } );
-
-    ee.on('finish', function () {
-      options.log( 'Finished ´appdmg´' );
-      callback();
-    } );
-
-    ee.on( 'error', function ( error ) {
-      callback( error );
-    } );
```
Comment (justinmchase) [inline lib/macos.js]: This entire red block was simply moved up above, verbatim, into the function `buildDarwin`.

Comment (stefanjudis) [inline lib/macos.js]: jup. :+1:

**Comment by stefanjudis** [comment]
I'll check the windows support now and set up travis properly.

After that I'd plan a minor release and push this one out afterwards.

Fine with that?

**Comment by justinmchase** [comment]
Sounds awesome :)

**Comment by justinmchase** [comment]
I messed up the remote branch for this pull request. Reopening as a new PR.

**Pull request closed without merging**